src/xcode/ENA/ENA/Source/HTTPClient/Resources/OTPAuthorizationForSRS/__tests__/OTPAuthorizationForSRSResourceTests.swift

```diff
@@ -0,0 +1,405 @@
+			}
+		}
+	}
+
+	func testGIVEN_AuthorizeOTP_WHEN_Failure_500StatusCode_THEN_internalServerErrorIsReturned() throws {
+		// GIVEN
+		let response: [String: String] = ["errorCode": "JWS_SIGNATURE_VERIFICATION_FAILED"]
+		let stack = MockNetworkStack(
+			httpStatus: 500,
+			responseData: try JSONEncoder().encode(response)
+		)
+
+		let otp = "OTPFake"
+		let ppacToken = PPACToken(apiToken: "APITokenFake", deviceToken: "DeviceTokenFake")
+
+		// WHEN
+		let serviceProvider = RestServiceProvider(session: stack.urlSession, cache: KeyValueCacheFake())
+		let resource = OTPAuthorizationForSRSResource(otpSRS: otp, ppacToken: ppacToken)
+		serviceProvider.load(resource) { result in
+			switch result {
+				// THEN
+			case .success:
+				XCTFail("success should not be called")
+			case .failure(let otpError):
+				guard case let .receivedResourceError(customError) = otpError else {
+					XCTFail("unexpected error case")
+					return
+				}
+				XCTAssertEqual(customError, .otherServerError)
+			}
+		}
+	}
+
+	func testGIVEN_AuthorizeOTP_WHEN_Failure_UnkownStatusCode_THEN_internalServerErrorIsReturned() throws {
+		// GIVEN
+		let response: [String: String] = ["errorCode": "JWS_SIGNATURE_VERIFICATION_FAILED"]
+		let stack = MockNetworkStack(
+			httpStatus: 91,
+			responseData: try JSONEncoder().encode(response)
+		)
+
+		let otp = "OTPFake"
+		let ppacToken = PPACToken(apiToken: "APITokenFake", deviceToken: "DeviceTokenFake")
+
+		// WHEN
+		let serviceProvider = RestServiceProvider(session: stack.urlSession, cache: KeyValueCacheFake())
+		let resource = OTPAuthorizationForSRSResource(otpSRS: otp, ppacToken: ppacToken)
+		serviceProvider.load(resource) { result in
+			switch result {
+				// THEN
+			case .success:
+				XCTFail("success should not be called")
+			case .failure(let otpError):
+				guard case let .receivedResourceError(customError) = otpError else {
+					XCTFail("unexpected error case")
+					return
+				}
+				XCTAssertEqual(customError, .otherServerError)
+			}
+		}
+	}
+
+	func testGIVEN_AuthorizeOTP_WHEN_Failure_JsonParsing_THEN_invalidResponseErrorIsReturned() {
+		// GIVEN
+		let stack = MockNetworkStack(
+			httpStatus: 400,
+			responseData: Data()
+		)
+
+		let otp = "OTPFake"
+		let ppacToken = PPACToken(apiToken: "APITokenFake", deviceToken: "DeviceTokenFake")
+
+		// WHEN
+		let serviceProvider = RestServiceProvider(session: stack.urlSession, cache: KeyValueCacheFake())
+		let resource = OTPAuthorizationForSRSResource(otpSRS: otp, ppacToken: ppacToken)
+		serviceProvider.load(resource) { result in
+			switch result {
+				// THEN
+			case .success:
+				XCTFail("success should not be called")
+			case .failure(let otpError):
+				guard case let .receivedResourceError(customError) = otpError else {
+					XCTFail("unexpected error case")
+					return
+				}
+				XCTAssertEqual(customError, .invalidResponseError)
+			}
+		}
+	}
+	
+	func testGIVEN_AuthorizeOTP_WHEN_NoNetworkConnection_THEN_NetworkErrorReturned() {
```

Comment:
Can you please add a test case for .deviceBlocked?

Reply:
Please refer to this PR: https://github.com/corona-warn-app/cwa-app-ios/pull/4928/files